third_party/helm/helm.go

```diff
@@ -0,0 +1,208 @@
+// does not have the ability to de-dup the arguments since options such as
+// -f or --set can be used multiple times with different values, and we wanted
+// the end user to make the decision of how the commands are invoked and not
+// restrict them to a specific way of invoking the commands.
+func WithArgs(args ...string) Option {
+	return func(opts *Opts) {
+		opts.Args = append(opts.Args, args...)
+	}
+}
+
+// WithWait is used to configure the argument used by the helm to indicate it
+// should wait for an acceptable state of the resource before yielding the
+// control back to the invocation point
+func WithWait() Option {
+	return func(opts *Opts) {
+		opts.Wait = true
+	}
+}
+
+// WithTimeout is used to configure the time for which the helm command should
+// be in blocked wait mode for the acceptable state of the resource to be reached
+func WithTimeout(timeout string) Option {
+	return func(opts *Opts) {
+		opts.Timeout = timeout
+	}
+}
+
+// processOpts is used to generate the Opts resource that will be used to generate
+// the actual helm command to be run using the getCommand helper
+func (m *Manager) processOpts(opts ...Option) *Opts {
+	option := &Opts{}
+	for _, op := range opts {
+		op(option)
+	}
+	return option
+}
+
+// getCommand is used to convert the Opts into a helm suitable command to be run
+func (m *Manager) getCommand(opt *Opts) (string, error) {
+	commandParts := []string{"helm", opt.Mode}
+	if opt.Mode == "" {
+		return "", fmt.Errorf("missing helm operation mode. Please use the WithMode option while invoking the Run")
+	}
+	if opt.Name != "" {
+		commandParts = append(commandParts, opt.Name)
+	}
+	if opt.Chart != "" {
+		commandParts = append(commandParts, opt.Chart)
+	} else {
+		commandParts = append(commandParts, opt.ReleaseName)
+	}
+	if opt.Namespace != "" {
+		commandParts = append(commandParts, "--namespace", opt.Namespace)
+	}
+	if opt.Version != "" {
+		commandParts = append(commandParts, "--version", opt.Version)
+	}
+	commandParts = append(commandParts, opt.Args...)
+	if opt.Wait {
+		commandParts = append(commandParts, "--wait")
+	}
+	if opt.Timeout != "" {
+		commandParts = append(commandParts, "--timeout", opt.Timeout)
+	}
+	commandParts = append(commandParts, "--kubeconfig", m.kubeConfig)
+	return strings.Join(commandParts, " "), nil
+}
+
+// Run method is used to invoke a helm command to perform a suitable operation.
+// Please make sure to configure the right Opts using the Option helpers
+func (m *Manager) Run(opts ...Option) (proc *exec.Proc, err error) {
+	if m.e.Prog().Avail("helm") == "" {
+		err = fmt.Errorf(missingHelm)
+		return
+	}
+	command, err := m.getCommand(m.processOpts(opts...))
+	if err != nil {
+		return
+	}
+	log.V(4).InfoS("Running Helm Operation", "command", command)
+	proc = m.e.RunProc(command)
+	return
+}
+
+func NewManager(kubeConfig string) *Manager {
+	return &Manager{e: gexe.New(), kubeConfig: kubeConfig}
+}
```

Comment:
How can we add/update the `helm` repo? 

```
helm repo add nginx-stable https://helm.nginx.com/stable
helm repo update
```

Reply:
```go
manager.Run(helm.WithMode("repo"), helm.WithArgs("add", "nginx-stable", "https://helm.nginx.com/stable"))
manager.Run(helm.WithMode("repo"), helm.WithArgs("update"))
```

This should let you run that workflow. However, I can write a simple wrapper around this to make the workflow much easier. However,. Do you have a list of these common invocation that can benifit from having a wrappers ?

Reply:
No need to write a wrapper for common ones, just wanted to make sure we have this feature.

Reply:
Sure. The `WithMode` method is provided so that you can invoke any helm sub command and use it with `WithArgs` to pass any non standard args that are already provided with a helpers.

Reply:
Can we add this in the example?

Reply:
Sure. I am doing that as we speak to address the suggestion from https://github.com/kubernetes-sigs/e2e-framework/pull/90#discussion_r766907634

Reply:
Done.